File: compiled_lang/cpp/src/f20.loop.cpp
// testbox: title="while loop"
#include <iostream>
#include <string>

int main() {
    std::string answer = "";
    while (answer != "quit") {
        std::cout << "Enter your name (quit to Exit): ";
        std::getline(std::cin, answer);

        if (answer != "quit")
            std::cout << "Hello " << answer << "!" << std::endl;
    }

    return 0;
}

File: compiled_lang/rust/f20.loop.rs
// testbox: title="while loop"
use std::io::{self, Write};

fn main() {
    let mut answer = String::new();
    while answer != "quit" {
        print!("Enter your name (quit to Exit): ");
        io::stdout().flush().unwrap();

        let mut line = String::new();
        io::stdin().read_line(&mut line).unwrap();
        answer = line.trim_end().to_string();

        if answer != "quit" {
            println!("Hello {}!", answer);
        }
    }
}
